docs/DeveloperGuide.md

```diff
@@ -250,24 +250,77 @@ The activity diagram shows the general sequence of steps when a user interacts w
 
 #### Implementation
 
+1. **Input Parsing**  
+   The user command input is parsed in `LogicManager` by calling `AddressBookParser#parseCommand`. For a command starting with "find person", the parser creates a `FindPersonCommandParser` instance to parse specific parameters.
+
+2. **Argument Tokenization**  
+   `FindPersonCommandParser#parse` uses `ArgumentTokenizer` to separate user input into arguments prefixed by indicators (e.g., `n/` for names). 
+
+3. **Predicate Creation**  
+   The `FindPersonCommandParser` checks if the `n/` prefix is present. If so, it creates a `NameContainsKeywordsPredicate` instance, containing a list of keywords derived from the user input, and passes it to `FindPersonCommand`.
+
+4. **Command Execution**  
+   Upon execution, `FindPersonCommand#execute` calls `Model#updateFilteredPersonList` to apply the predicate, filtering the list of persons based on keyword matches. The command then constructs a success message indicating the number of persons found.
+
+5. **Error Handling**
+    - If the required prefix (e.g., `n/`) is missing, the parser throws a `ParseException`.
+    - If an `IOException` or `AccessDeniedException` occurs during data storage, `LogicManager` handles these with an appropriate `CommandException`.
+   
 #### Design considerations
 
+### 1. Predicate Composition
+- **Reason**: Using predicates enables flexible filtering and reduces redundancy by allowing shared filtering logic across commands.
+
+### 2. Case-Insensitive Matching
+- **Reason**: User experience is enhanced by making searches case-insensitive.
+
+
 <br>
 
 ### List person feature
 
 #### Implementation
 
+1. **Input Parsing**  
+   The `list` command input is parsed in `LogicManager` through the `AddressBookParser#parseCommand` method. When the command starts with `list`, the parser invokes `ListCommandParser` to handle the specific arguments.
+
+2. **Entity Type Verification**  
+   `ListCommandParser#parse` trims the input to verify the entity type, checking for `PERSON_ENTITY_STRING` or `APPOINTMENT_ENTITY_STRING`.
+   If the entity type is `PERSON_ENTITY_STRING`, `ListCommandParser` returns an instance of `ListPersonCommand`.
```

Comment:
This is clear, but should we include it at the start of the section, since it applies to all the commmandparsers?

Reply:
This too.